src/telegrafs/actions/thunks.ts

```diff
@@ -108,6 +111,61 @@ export const createTelegraf = (telegraf: Telegraf) => async (
   }
 }
 
+const cloneTelegrafLabels = (
+  sourceTelegraf: Telegraf,
+  destinationTelegraf: Telegraf
+) => async (dispatch: Dispatch<Action>) => {
+  try {
+    const pendingLabels = sourceTelegraf.labels.map(labelID =>
+      postTelegrafsLabel({
+        telegrafID: destinationTelegraf.id,
+        data: {labelID},
+      })
+    )
+
+    const mappedLabels = await Promise.all(pendingLabels)
+
+    if (
+      mappedLabels.length &&
+      mappedLabels.some(label => label.status !== 201)
+    ) {
+      throw new Error(
+        'An error occurred cloning the labels for this telegraf config'
+      )
+    }
+    dispatch(notify(cloneTelegrafLabelsSuccess()))
+  } catch {
+    dispatch(notify(addTelegrafLabelFailed()))
+  }
+}
+
+export const cloneTelegraf = (telegraf: Telegraf) => async (
+  dispatch: Dispatch<Action>
+) => {
+  let clonedTelegraf
+
+  // Step 1: create a new telegraf
+  try {
+    const response = await postTelegraf({data: telegraf})
+
+    if (response.status !== 201) {
+      throw new Error(response.data.message)
+    }
+
+    clonedTelegraf = response.data
+    dispatch(notify(cloneTelegrafSuccess()))
+  } catch (error) {
+    console.error(error)
+    dispatch(notify(telegrafCreateFailed()))
+  }
+
+  // Step 2: clone the labels
+  cloneTelegrafLabels(telegraf, clonedTelegraf)(dispatch)
```

Comment:
I think you can also do:

```ts
dispatch(cloneTelegrafLabels(telegraf, clonedTelegraf))
```

Reply:
oh nice, yeah that's better. it's consistent. i will make the change.

Reply:
oops. no, unfortunately we can't. i just tried it and TypeScript is telling i am passing in the wrong thing to `dispatch`.